chain/client/src/stateless_validation/state_witness_distribution_actions.rs

```diff
@@ -0,0 +1,29 @@
+use near_async::messaging::CanSend;
+use near_chain::Error;
+use near_network::types::{NetworkRequests, PeerManagerAdapter, PeerManagerMessageRequest};
+
+use super::state_witness_distribution_actor::DistributeChunkStateWitnessRequest;
+
+#[derive(Clone)]
+pub struct StateWitnessDistributionActions {
```

Comment:
I see this pattern elsewhere, but what is the motivation to have a separate actions struct separate from the actor? can we simply move this to the distribution actor instead?

Reply:
So this one is complex and I don't fully understand it but perhaps @robin-near can shed a bit more light on this.

This is related to the test_loop code that we have. We can't have actix actors but we would like to create handlers for the messages. The way of doing this is to have functions like `forward_sync_jobs_messages_from_client_to_sync_jobs_actions` that you see, which take in an action object (which is basically an abstraction of the logic of the actor) and keeping a copy of it for the test loop running.

I'm not a fan of this patter and would really love to simplify it if there's a way.

Reply:
ok, I see that this structure is allowing working with actions without existence of an actor, as in the test cases.